catalog/core/catalog-core-api/src/main/java/ddf/catalog/CatalogFramework.java

```diff
@@ -70,14 +72,43 @@
  * {@link ddf.catalog.transform.QueryResponseTransformer}s.
  * </p>
  *
- * @author Michael Menousek, Lockheed Martin
- * @author Ashraf Barakat, Lockheed Martin
- * @author Ian Barnett, Lockheed Martin
- * @author ddf.isgs@lmco.com
  */
 public interface CatalogFramework extends Describable {
 
     /**
+     * <p>
+     * <b> This code is experimental. While this interface is functional and tested, it may change or be
+     * removed in a future version of the library. </b>
+     * </p>
+     *
+     * Creates {@link Metacard}s in the {@link ddf.catalog.source.CatalogProvider}.
+     *
+     * <p>
+     * <b>Implementations of this method must:</b>
+     * <ol>
+     * <li/>Before evaluation, call {@link ddf.catalog.plugin.PreIngestPlugin#process(CreateRequest)} for each
+     * registered {@link ddf.catalog.plugin.PreIngestPlugin} in order determined by the OSGi SERVICE_RANKING
+     * (Descending, highest first), "daisy chaining" their responses to each other.
+     * <li/>Call {@link ddf.catalog.source.CatalogProvider#create(CreateRequest)} on the registered
```

Comment:
Those plugins are called, but they are called after all of the original catalog plugins. That also includes the security plugins so that the metadata itself can be evaluated before any of the storage operations are executed.